charts/rawls/templates/migration/role.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.migration.enabled }}
```

Comment:
Do we need separate RBAC resources for this job? I think the permissions will be very similar to whatever the Rawls application uses. (Apologies if there's already been some background discussion here that I missed!)

Seems like we could add PreSync to the existing resources to make sure they exist when the job is run, but not delete them before every sync?

Reply:
We don't strictly need separate RBAC, but we do need them to exist, and unfortunately: https://github.com/argoproj/argo-cd/issues/3502#issuecomment-620808412

> Argo CD hooks are not considered part of Argo CD application resources, and features like drift detection and health assessments do not work. So the solution should involve not having to resort to adding `"helm.sh/hook": pre-install` to a Namespace resource.

Despite the clutter that this involves (https://github.com/broadinstitute/terra-helm/pull/387#discussion_r651287596) I don't know of a better solution

Reply:
Does this comment offer a viable workaround? https://github.com/argoproj/argo-cd/issues/3502#issuecomment-620832168

I.e.
* RBAC resources sync in wave -2
* Job syncs in wave -1 with a Sync hook
* Everything else syncs in default wave 0

If that won't work, I'm fine with duplicating the RBAC resources!

Reply:
I hadn't seen that, that's a good idea. I'd talked to Mike and just now put the RBAC stuff behind a flag so the job can reuse the existing one. I think the only downside to leaning on sync waves is that the normal rbac resources would need to have that sort of migration-specific attribute, but that's a small downside imo. Let's chat at standup

Reply:
This worked!